csrc/kernel_cache.cpp

```diff
@@ -1337,7 +1344,9 @@ void FusionKernelRuntime::compileKernel(
     fusion_to_run->printMath();
   }
   FusionGuard fg(fusion_to_run.get());
-  scheduler_entry->schedule(fusion_to_run.get());
+  if (auto_schedule) {
```

Comment:
Is this the only place disabled when `auto_schedule` is false?

Reply:
yes, this is the only place where this flag is used, otherwise it is just passed around through the interfaces.

Reply:
Including the segmentation?

Reply:
yes. The segmentation happens before.

Reply:
I'm trying to understand what `auto_schedule` means here since the segmentation is effectively almost part of the scheduling.

I think what's not very clear is what is assumed to be done with the given fusion. Since `schedule` is not called, the fusion is assumed to be scheduled, but since the segmenter kicks in, I'm a bit surprised that the segmenter actually works with a pre-scheduled fusion.

Reply:
> I'm a bit surprised that the segmenter actually works with a pre-scheduled fusion.

The segmenter works because the error we might encounter with a pre-scheduled fusion happens down the line when we want to actually apply the scheduling.

So, in this patch, we "dupe" the segmenter: it thinks each segment will be scheduled wit ha given heuristic but we actually bypass the application of that heuristic